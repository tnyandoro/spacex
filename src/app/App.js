import React from 'react';
import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
import Missions from '../features/missions/Missions';
import NavBar from '../features/navBar/NavBar';
import Profile from '../features/profile/Profile';
import Rockest from '../features/rockets/Rockets';
import './app.css';

const routes = [
  {
    name: 'Rockest',
    path: '/',
    component: <Rockest />,
  },
  {
    name: 'Missions',
    path: '/missions',
    component: <Missions />,
  },
  {
    name: 'My Profile',
    path: '/profile',
    component: <Profile />,
  },
];

function App() {
  return (
    <Router basename={process.env.PUBLIC_URL}>
      <div className="app">
        <NavBar title="Space Travelers' Hub" routes={routes} />
        <Routes>
          { routes.map(
            ({ path, component }) => (
              <Route key={path} path={path} element={component} />
            ),
          )}
        </Routes>
      </div>
    </Router>
  );
}

export default App;
