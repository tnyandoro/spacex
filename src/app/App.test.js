import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import App from './App';

it('App snapshot test', () => {
  const initialState = {
    rocketsReducer: [],
    missionsReducer: [],
  };
  const mockStore = configureStore();
  const store = mockStore(initialState);

  const tree = renderer
    .create(
      <Provider store={store}>
        <App />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
