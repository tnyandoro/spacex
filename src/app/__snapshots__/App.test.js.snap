// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot test 1`] = `
<div
  className="app"
>
  <nav
    className="navBar"
  >
    <ul
      className="nav-list"
    >
      <li
        className="brand"
      >
        <a
          className="nav-brand"
          href="/"
          onClick={[Function]}
        >
          <img
            alt="logo"
            className="App-logo"
            src="planet.png"
          />
          <h1
            className="nav-title"
          >
            Space Travelers' Hub
          </h1>
        </a>
      </li>
      <li
        className="nav-links"
      >
        <a
          aria-current="page"
          className="links active"
          exact="true"
          href="/"
          onClick={[Function]}
        >
          Rockest
        </a>
      </li>
      <li
        className="nav-links"
      >
        <a
          className="links"
          exact="true"
          href="/missions"
          onClick={[Function]}
        >
          Missions
        </a>
      </li>
      <li
        className="nav-links"
      >
        <a
          className="links"
          exact="true"
          href="/profile"
          onClick={[Function]}
        >
          My Profile
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="rocket-wrapper"
  >
    <ul
      className="ul-rocket"
    />
  </div>
</div>
`;
